fh_immuta_utils/client.py

```
@@ -294,17 +293,35 @@ def create_data_source(
             f"{request_prefix}/handler", json=post_body
         )
         self.log.debug("Response: %s", handler_response.text)
-        # There's no ID given back for bulk create requests
         if handler_response.status_code == 200:
-            if not is_bulk_insert:
-                return self.get_data_source(handler_response.json()["dataSourceId"])
-            return None
+            return handler_response.json()
         if "already exists" in handler_response.text:
             self.log.info("Data source with name %s already exists", data_source.name)
             return None
         self.log.error("Error: %s", handler_response.text)
         handler_response.raise_for_status()
 
+    def wait_for_data_source_creation_completion(
+        self, connection_string: str, check_interval: int = 30
+    ) -> None:
+        """
+        Returns when the jobs for creating data sources with connection string `connection_string`
+        are complete. Checks the condition once every `check_interval` seconds.
+        """
+        while True:
+            try:
+                response = self.get(f"jobs?connectionString={connection_string}")
+                n_pending_jobs = int(response["pending"])
+                if n_pending_jobs == 0:
+                    return
+                else:
+                    self.log.info(
+                        f"{n_pending_jobs} jobs remaining for {connection_string}"
+                    )
+            except HTTPError:
```

is this the right class of errors to swallow and blindly retry on?
i.e. if our request is not well formed/theres an api change?

---

We should at least log this error

---

not sure why im catching the error tbh, i think it was for debugging - removed the try/except